app/scripts/modules/core/src/presentation/forms/fields/ExpressionRegexField.tsx

```diff
@@ -0,0 +1,118 @@
+import { CollapsibleSection } from 'core/presentation';
+import { ValidationError, ValidationWarning } from 'core/validation';
+import * as React from 'react';
+
+import { ExpressionError, ExpressionInput, ExpressionPreview, ISpelError } from '../inputs';
+import { IFieldLayoutProps } from '../interface';
+import { BasicLayout } from '../layouts';
+import { IExpressionFieldProps } from './ExpressionField';
+import { formikField, IFormikField } from './formikField';
+
+export interface IRegexpProps {
+  RegexHelp?: JSX.Element;
+  regex: string;
+  replace: string;
+  onRegexChanged: (regex: string) => void;
+  onReplaceChanged: (replace: string) => void;
+}
+
+export type IExpressionRegexFieldProps = IRegexpProps & IExpressionFieldProps;
+
+export interface IExpressionRegexFieldState {
+  spelPreview: string;
+  spelError: ISpelError;
+}
+
+export class ExpressionRegexField extends React.Component<IExpressionRegexFieldProps, IExpressionRegexFieldState> {
+  static Formik: IFormikField<IExpressionRegexFieldProps> = formikField<IExpressionRegexFieldProps>(
+    ExpressionRegexField,
+  );
+  static defaultProps: Partial<IExpressionFieldProps> = { markdown: false, FieldLayout: BasicLayout };
+  public state: IExpressionRegexFieldState = {
+    spelPreview: null,
+    spelError: null,
+  };
+
+  private renderRegexFields(props: IExpressionRegexFieldProps) {
+    const { RegexHelp, regex, replace, onRegexChanged, onReplaceChanged } = props;
+
+    return (
+      <CollapsibleSection
+        heading={({ chevron }) => (
+          <span>
+            {chevron} Regex {RegexHelp}
+          </span>
+        )}
+        outerDivClassName=""
+        toggleClassName="clickable"
+        defaultExpanded={!!regex}
+      >
+        <div className="flex-container-h baseline">
+          <code>s/</code>
+          <input
+            type="text"
+            value={regex || ''}
+            onChange={evt => onRegexChanged(evt.target.value)}
+            className="flex-grow"
```

Comment:
easier if we set it at the layout level so depending on the layout, it sets these on all input fields.